### PR TITLE
Add utility bill calculations

### PR DESCRIPTION
**Pull request by asparke2**
Pull request overview
---------------------

This pull request adds utility bill calculations for electricity, natural gas, propane, and fuel oil. Electricity rates are from the U.S. Utility Rate Database on OpenEI. Petroleum fuel rates are volumetric prices derived from EIA data. All bills reflect 2022 prices.
 
See change documentation here: https://nrel.sharepoint.com/:w:/r/sites/comstock/Shared%20Documents/ComStock%20Filing%20Cabinet/ComStock%20Core/Change%20Documentation/Changes/vTBD%20Utility%20Rates/ComStock%20Change%20Adds%20Utility%20Rates.docx?d=w1dbdbc340bec4de9a6234a3d05a9345c&csf=1&web=1&e=teBIKO

### Pull Request Author

This pull request makes changes to (select all the apply):
 - [X] Documentation
 - [ ] Infrastructure (includes singularity image, buildstock batch, dependencies, continuous integration tests)
 - [ ] Sampling
 - [ ] Workflow Measures
 - [ ] Upgrade Measures
 - [X] Reporting Measures
 - [X] Postprocessing

Author pull request checklist:

 - [X] Tagged the pull request with the appropriate label (documentation, infrastructure, sampling, workflow measure, upgrade measure, reporting measure, postprocessing) to help categorize changes in the release notes.
 - [X] Added tests for new measures
 - [X] Updated measure .xml(s)
 - [X] Register values added to `comstock_column_definitions.csv`
 - [ ] Both `options_lookup.tsv` files updated
 - [X] 10k+ test run
 - [X] Change documentation written
 - [X] Measure documentation written
 - [X] ComStock documentation updated
 - [ ] Changes reflected in example `.yml` files
 - [ ] Changes reflected in `README.md` files
 - [X] Added 'See ComStock License' language to first two lines of each code file
 - [ ] All new and existing tests pass the CI

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [x] Perform a code review on GitHub
 - [ ] All related changes have been implemented: data and method additions, changes, tests
 - [ ] If fixing a defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [x] Reviewed change documentation
 - [x] Ensured code files contain License reference
 - [x] Results differences are reasonable
 - [x] CI status: all tests pass

#### ComStock Licensing Language - Add to Beginning of Each Code File
```
# ComStock™, Copyright (c) 2023 Alliance for Sustainable Energy, LLC. All rights reserved.
# See top level LICENSE.txt file for license terms.
```

### CODE REVIEW

File: measures/utility_bills/resources/calc_elec_bill.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+parser.add_argument("kwhpath", help="Full path to kwh CSV file")
+parser.add_argument("ratepath", help="Full path to URDB rate JSON file")
+args = parser.parse_args()
+
+# Load the hourly kWh consumption from disk
+hourly_kwh = []
+with open(args.kwhpath) as f:
+  for val in f:
+    hourly_kwh.append(float(val))
+assert len(hourly_kwh) == 8760, f"Got {len(hourly_kwh)} instead of expected 8760 hourly kWh values"
```
Comment (mdahlhausen) [inline measures/utility_bills/resources/calc_elec_bill.py:28]: how will you account for leap years?

File: measures/utility_bills/measure.rb [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+  end
+
+  # return a vector of IdfObject's to request EnergyPlus objects needed by the run method
+  def energyPlusOutputRequests(runner, user_arguments)
+    super(runner, user_arguments)
+
+    result = OpenStudio::IdfObjectVector.new
+
+    # Request hourly data for fuel types with hourly bill calculations
+    fuel_types = ['Electricity'] # 'NaturalGas', 'DistrictCooling', 'DistrictHeating', 'FuelOilNo2', 'Propane'
```
Comment (mdahlhausen) [inline measures/utility_bills/measure.rb]: why are other fuel types commented out?

**Review by mdahlhausen — DISMISSED** [review]

**Comment by mdahlhausen** [comment]
Other comments:
- Add a measure description in the [meta and reporting measure README](https://github.com/NREL/ComStock/blob/main/measures/README.md)

**Review by eringold — DISMISSED** [review]

File: measures/utility_bills/resources/calc_elec_bill.py (hunk 1 above)
Comment (asparke2) [reply measures/utility_bills/resources/calc_elec_bill.py:28]: ComStock runs the first 365 days of the year for leap years, so it cuts off Dec 31st when there is a leap year, always leaving 8760 hours worth of timeseries data

File: measures/utility_bills/measure.rb (hunk 2 above)
Comment (asparke2) [reply measures/utility_bills/measure.rb]: We only need hourly data for electricity bill calculations, the other fuels are all currently annual volumetric prices.

**Comment by asparke2** [comment]
@eringold @mdahlhausen made another two fixes and ran:
1. Actually pull the start date from each rate and use that as the basis for adjustment to 2022
2. Add code to shift the timeseries electricity data to match a Monday start day (PySAM's assumption). This didn't affect the rates since ComStock uses 2018 which happens to start on a Monday, but covered for future/good measure.

Result:
Electricity rates generally dropped a bit. This helped the most over-estimated region, driven by CA utilities who have more up-to-date rates in the URDB. Updated the change doc with new figures if you want to look. Otherwise I think this is good to merge.

**Review by eringold — APPROVED** [review]

**Pull request merged**